graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfileService.java

```diff
@@ -77,31 +77,31 @@ public class IndexFieldTypeProfileService extends PaginatedDbService<IndexFieldT
             .sort(Sorting.create(IndexFieldTypeProfile.NAME_FIELD_NAME, Sorting.Direction.valueOf("asc".toUpperCase(Locale.ROOT))))
             .build();
 
-    private final MongoCollection<IndexFieldTypeProfile> profileCollection;
+    private final MongoCollection<IndexFieldTypeProfile> collection;
     private final DbQueryCreator dbQueryCreator;
     private final IndexFieldTypeProfileUsagesService indexFieldTypeProfileUsagesService;
     private final IndexSetService indexSetService;
+    private final MongoUtils<IndexFieldTypeProfile> mongoUtils;
 
     @Inject
-    public IndexFieldTypeProfileService(final MongoConnection mongoConnection,
-                                        final MongoJackObjectMapperProvider mapper,
-                                        final MongoCollections mongoCollections,
+    public IndexFieldTypeProfileService(final MongoCollections mongoCollections,
                                         final IndexFieldTypeProfileUsagesService indexFieldTypeProfileUsagesService,
                                         final IndexSetService indexSetService) {
-        super(mongoConnection, mapper, IndexFieldTypeProfile.class, INDEX_FIELD_TYPE_PROFILE_MONGO_COLLECTION_NAME);
-        this.db.createIndex(new BasicDBObject(IndexFieldTypeProfile.NAME_FIELD_NAME, 1), new BasicDBObject("unique", false));
-        this.profileCollection = mongoCollections.collection(INDEX_FIELD_TYPE_PROFILE_MONGO_COLLECTION_NAME, IndexFieldTypeProfile.class);
         this.indexSetService = indexSetService;
         this.dbQueryCreator = new DbQueryCreator(IndexFieldTypeProfile.NAME_FIELD_NAME, ATTRIBUTES);
         this.indexFieldTypeProfileUsagesService = indexFieldTypeProfileUsagesService;
+
+        collection = mongoCollections.collection(INDEX_FIELD_TYPE_PROFILE_MONGO_COLLECTION_NAME, IndexFieldTypeProfile.class);
+        mongoUtils = mongoCollections.utils(collection);
+
+        collection.createIndex(Indexes.ascending(IndexFieldTypeProfile.NAME_FIELD_NAME));
```

Comment:
Originally index was unique. It seems that you have forgotten about it when refactoring.

Reply:
🤔 was it? It looks like it was explicitly set to not unique, which I think is the default. Or am I missing something?

Reply:
Oh, sorry, my mistake...